crates/scarb-artifacts/src/lib.rs

```diff
@@ -164,24 +171,20 @@ pub fn corelib_for_package(metadata: &Metadata, package: &PackageId) -> Result<U
     Ok(Utf8PathBuf::from(corelib.source_root()))
 }
 
-/// Get the top-level and main file paths for the given package
+/// Get the package root directory and manifest file paths for the given package
 pub fn paths_for_package(
     metadata: &Metadata,
     package: &PackageId,
 ) -> Result<(Utf8PathBuf, Utf8PathBuf)> {
-    let compilation_unit = compilation_unit_for_package(metadata, package)?;
-
     let package = metadata
         .get_package(package)
         .ok_or_else(|| anyhow!("Failed to find metadata for package = {package}"))?;
 
-    let package_path = package.root.clone();
-    let package_source_dir_path = compilation_unit.target.source_root();
-
-    Ok((package_path, Utf8PathBuf::from(package_source_dir_path)))
+    Ok((package.root.clone(), package.manifest_path.clone()))
```

Comment:
Was this only used in forge? So we don't accidentally break some behavior in cast.

Reply:
Yup, it was only used in forge